src/core/data/particles/particle_array.hpp

```diff
@@ -230,6 +230,22 @@ class ParticleArray
     auto& vector() { return particles_; }
     auto& vector() const { return particles_; }
 
+    auto& update_from(This const& that)
+    {
+        bool resize_bigger = this->size() < this->size();
+        if (resize_bigger)
+            this->resize(this->size() + static_cast<std::size_t>(this->size() * .05));
```

Comment:
I'm wondering about this 0.05 over-allocation coef. How does it compare to the size of 1 element (1 particle?)
Also we could write it `(1+coef)*this->size()`?

Reply:
> I'm wondering about this 0.05 

me too

> How does it compare to the size of 1 element (1 particle?)

Assuming it floors it would be 1+0 (if I get what you meant)

> Also we could write it (1+coef)*this->size()?

sure

the idea here (which you probably got), is that we might get an array in another patch that has like one or two more particles, and we'd have have to reallocatate again if we didn't over allocate previously, so trying to avoid that, while also not taking too much additional memory